src/main/java/techreborn/blockentity/machine/tier1/GreenhouseControllerBlockEntity.java

```diff
@@ -0,0 +1,241 @@
+			return false;
+		}
+		for (int i = -1; i <= 1; i++) {
+			for (int j = -1; j <= 1; j++) {
+				BlockState block = world.getBlockState(multiblockCenter.add(i * 3, 3, i * 3));
+				if (!(block.getBlock() instanceof BlockLamp) || !BlockLamp.isActive(block)) {
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+	
+	@Override
+	public void tick() {
+		if (multiblockCenter == null) {
+			multiblockCenter = pos.offset(getFacing().getOpposite(), 5);
+		}
+		charge(6);
+		super.tick();
+		
+		if (world.isClient) {
+			return;
+		}
+		
+		if (useEnergy(getEuPerTick(TechRebornConfig.greenhouseControllerEnergyPerTick)) != getEuPerTick(TechRebornConfig.greenhouseControllerEnergyPerTick)) {
+			return;
+		}
+		
+		if (--ticksToNextMultiblockCheck < 0) {
+			growthBoost = getMultiBlock();
+			ticksToNextMultiblockCheck = 200;
+		}
+		
+		if (world.getTime() % 20 == 0) {
+			double cyclesLimit = getSpeedMultiplier() * 4 + 1;
+			while (cyclesLimit-- > 0) {
+				workCycle();
+			}
+		}
+		
+	}
+	
+	private void workCycle() {
+		BlockPos blockPos = multiblockCenter.add(world.random.nextInt(9) - 4, 0, world.random.nextInt(9) - 4);
+		BlockState blockState = world.getBlockState(blockPos);
+		Block block = blockState.getBlock();
+		
+		if (growthBoost) {
+			if (block instanceof Fertilizable
+					|| block instanceof PlantBlock
+					|| block instanceof SugarCaneBlock
+					|| block instanceof CactusBlock
+			) {
+				if (canUseEnergy(TechRebornConfig.greenhouseControllerEnergyPerBonemeal)) {
+					useEnergy(TechRebornConfig.greenhouseControllerEnergyPerBonemeal);
+					blockState.scheduledTick((ServerWorld) world, blockPos, world.random);
+				}
+			}
+		}
+		
+		if (block instanceof CropBlock) {
+			processAgedCrop(blockState, blockPos, ((CropBlock) block).getAgeProperty(), ((CropBlock) block).getMaxAge(), 0);
+		} else if (block instanceof NetherWartBlock) {
+			processAgedCrop(blockState, blockPos, NetherWartBlock.AGE, 3, 0);
+		} else if (block instanceof SweetBerryBushBlock) {
+			processAgedCrop(blockState, blockPos, SweetBerryBushBlock.AGE, 3, 1);
+		} else if (block instanceof CocoaBlock) {
+			processAgedCrop(blockState, blockPos, CocoaBlock.AGE, 2, 0);
+		} else if (block instanceof GourdBlock) {
+			if (tryHarvestBlock(blockState, blockPos)) {
+				world.breakBlock(blockPos, false);
+			}
+		} else if (block instanceof SugarCaneBlock
+				|| block instanceof CactusBlock
+				|| block instanceof BambooBlock
+		) {
+			// If we can break bottom block we should at least remove all of them up to top so they don't break automatically
+			boolean breakBlocks = false;
+			for (int y = 1; (blockState = world.getBlockState(blockPos.up(y))).getBlock() == block; y++) {
+				if (y == 1) {
+					breakBlocks = tryHarvestBlock(blockState, blockPos.up(y));
+				} else {
+					tryHarvestBlock(blockState, blockPos.up(y));
+				}
+				if (breakBlocks) world.breakBlock(blockPos.up(y), false);
+			}
+		} else if (block instanceof BlockRubberLog) {
```

Comment:
For now this can stay here, not sure if it will be better with a tree chopping block or not.